src/database/database.py

```diff
@@ -1897,3 +1884,231 @@ def get_friend_requests_for_target_user(self, target: str):
         finally:
             session.close()
         return result
+
+    def add_recipes(self, recipes: list):
+        """
+        This is a convenience function for adding recipes in bulk.
+        This is the same as making repeated calls to `add_recipe`, but this function is faster.
+
+        Args:
+            recipes (list): The list of recipes to add. Each recipe must be a JSON object containing the following values:
+                id (int): The ID of the recipe.
+                name (str): The name of the recipe.
+                image (str): The image URL of the recipe.
+                summary (str): The summary of the recipe. This value is optional.
+                full_summary (str): The full summary of the recipe. This value is optional.
+        """
+
+        from database.models import Recipe
+
+        actual_recipes = []
+
+        for recipe in recipes:
+            if self.get_recipe(recipe["id"]) != None:
+                raise DuplicateRecipeException(recipe["id"])
+            summary = ""
+            full_summary = ""
+            if "summary" in recipe.keys():
```

Comment:
I believe the spoonacular function tackles the error for it to always have a summary, even if it short and made up. This check would be good for when the JSON data is returned raw.